pkg/auth/authenticator/password/ldappassword/ldaputil/getattribute.go

```diff
@@ -0,0 +1,29 @@
+package ldaputil
+
+// getValue returns the first non-empty value the entry has for the given attributes
+import (
+	"strings"
+
+	"github.com/go-ldap/ldap"
+)
+
+// GetAttributeValue finds the first attribute of those given that the LDAP entry has, and
+// returns it. GetAttributeValue is able to query the DN as well as Attributes of the LDAP entry.
+// If no value is found, the empty string is returned.
+func GetAttributeValue(entry *ldap.Entry, attributes []string) string {
```

Comment:
This is used in the `ldaputil` package but also in quite a few of the LDAP-interacting portions of the LDAP sync. Where is it better off living?